tests/test_functional.py

```diff
@@ -432,3 +433,76 @@ def test_query_users():
     query = UsersQuery(conn, in_group="test", in_domain="test.com", direct_only=False)
     assert query.url_params == ["test"]
     assert query.query_params == {"directOnly": False, "domain": "test.com"}
+
+
+def test_large_group_assignment_split():
```

Comment:
I know the group_splitting code is in the functional file, but that doesn't really mean it has to be tested in a functional test.  The functional tests are really about making sure the functional wrappers are sending in the right stuff to the underlying generic APIs.  I think it would be better to move all these tests into the connections file, because they are really about logic that gets executed by the connections code.